content/200-concepts/100-components/02-prisma-client/260-client-extensions/100-model.mdx

```diff
@@ -133,3 +133,110 @@ prisma.$extends({
 ```
 
 You cannot use `Prisma.getExtensionContext(this)` inside an [`$allModels`](#add-a-custom-method-to-all-models-in-your-schema) structure.
+
+## Advanced type safety on models
+
+<Admonition type = "info">
+
+This feature is available from version 4.9.0.
+
+</Admonition>
+
+**Rewrite this intro on Monday:** To help with type safety, Prisma provides code validation for the internal types in generic models. It is especially useful for published extensions, because other users can more easily benefit from autocompletion, no matter what models apply. **TBA:** link to "sharing extensions" section; also link here from that section, but make it clear in the link text that this is an advanced topic
```

Comment:
To help with authoring extensions that are highly type-safe, the Prisma Client provides a set of type utilities to be able to tap into input and output types. These are fully dynamic, which means they will adapt to any given model and schema, helping you deliver a great auto-completion and developer experience for your extension.

Reply:
Thanks, I've brought this text into the intro